src/compiler/checker.ts

```diff
@@ -10547,62 +10547,102 @@ module ts {
             return false;
         }
 
-        function getSymbolsInScope(location: Node, meaning: SymbolFlags): Symbol[] {
+        function getSymbolsInScope(location: Node, meaning: SymbolFlags, predicate?: (symbol: Symbol) => boolean): Symbol[] {

```

Comment:
Consider adding some variation of the following comment:

```TypeScript
/**
 * Aggregates all symbols visible at a given location node. If two symbols with the same name exist
 * in different scopes, then the symbol whose declaration is lexically closer is returned.
 *
 * @param location   The node at which the client would like to get available symbols
 * @param meaning    A flag to determine which kinds of symbols to include.
 *                     If a symbol satisfies a flag, it will be considered.
 * @param predicate  If provided and any symbol satisfies this predicate, then a singleton list
 *                     consisting of that element is returned.
 */
```